src/librustdoc/formats/cache.rs

```diff
@@ -131,44 +130,22 @@ struct CacheBuilder<'a, 'tcx> {
 }
 
 impl Cache {
-    crate fn from_krate<'tcx>(
-        render_info: RenderInfo,
-        document_private: bool,
+    crate fn new(access_levels: AccessLevels<DefId>, document_private: bool) -> Self {
+        Cache { access_levels, document_private, ..Cache::default() }
+    }
```

Comment:
This doesn't work because the fields are private.
```
error[E0451]: field `stack` of struct `Cache` is private
   --> src/librustdoc/core.rs:525:92
    |
525 |         cache: Cache { access_levels, document_private: render_options.document_private, ..Cache::default() },
    |                                                                                            ^^^^^^^^^^^^^^^^ field `stack` is private
error[E0451]: field `parent_stack` of struct `Cache` is private
error[E0451]: field `parent_is_trait_impl` of struct `Cache` is private
error[E0451]: field `stripped_mod` of struct `Cache` is private
error[E0451]: field `masked_crates` of struct `Cache` is private
error[E0451]: field `orphan_trait_impls` of struct `Cache` is private
error: aborting due to 6 previous errors

Reply:
Hm, I didn't realize `..default()` worked that way—I would have assumed the `Default` impl could set private fields. I wonder why it works this way.